pkg/config/v3/config.go

```diff
@@ -29,15 +29,38 @@ import (
 // Version is the config.Version for project configuration 3
 var Version = config.Version{Number: 3}
 
+// stringSlice is a []string but that can also be unmarshalled from a single string,
+// which is introduced as the first and only element of the slice
+// It is used to offer backwards compatibility as the field used to be a string.
+type stringSlice []string
+
+func (ss *stringSlice) UnmarshalJSON(b []byte) error {
+	if b[0] == '[' {
+		var sl []string
+		if err := yaml.Unmarshal(b, &sl); err != nil {
+			return err
+		}
+		*ss = sl
+		return nil
+	}
+
+	var st string
+	if err := yaml.Unmarshal(b, &st); err != nil {
+		return err
+	}
+	*ss = stringSlice{st}
```

Comment:
This is supposedly a comma-separated list right? Also no need to unmarshal.
```suggestion
	*ss = stringSlice(strings.Split(string(b), ","))
```

Reply:
It is not a comma separated list, plugin phase 1.5 planned on suign a comma separated list to keep it as a string, but it is a single string at the moment.

We either need to unmarshal or to handle quotes ourself, as `strings.Unquote` requires them to be there while we may not have them so unmarshalling is just the easist way to get it done.

Reply:
Wasn't the whole point of `stringSlice` to support both a comma separated list and a YAML list?

Reply:
My point is, if there was no previous support for comma separated list, and we are going to use a YAML array after this, why do we need to support a comma separated list? Any time a command is run the config file will be rewritten to the YAML array format, the reason to support a string is backwards compatibility, but comma separated list has never been there so I don't see any reason to support it.

Reply:
I see your point. String splitting guards against someone accidentally using a comma separated value, but perhaps it would be better to error out in that case.